mcs/restapi/admin_handlers_test.go

```diff
@@ -0,0 +1,89 @@
+// This file is part of MinIO Kubernetes Cloud
+// Copyright (c) 2019 MinIO, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package restapi
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/minio/minio/pkg/madmin"
+
+	"errors"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// assigning mock at runtime instead of compile time
+var minioListUsersMock func() (map[string]madmin.UserInfo, error)
+
+// Define a mock struct of Admin Client interface implementation
+type adminClientMock struct {
+}
+
+// mock function of listUsers()
+func (ac adminClientMock) listUsers() (map[string]madmin.UserInfo, error) {
+	return minioListUsersMock()
+}
+
+func TestListUsers(t *testing.T) {
+	assert := assert.New(t)
+	adminClient := adminClientMock{}
+	// Test-1 : listUsers() Get response from minio client with two users and return the same number on listUsers()
+	// mock minIO client
+	mockUserMap := map[string]madmin.UserInfo{
+		"ABCDEFGHI": madmin.UserInfo{
+			SecretKey:  "",
+			PolicyName: "ABCDEFGHI-policy",
+			Status:     "enabled",
+			MemberOf:   []string{"group1", "group2"},
+		},
+		"ZBCDEFGHI": madmin.UserInfo{
+			SecretKey:  "",
+			PolicyName: "ZBCDEFGHI-policy",
+			Status:     "enabled",
+			MemberOf:   []string{"group1", "group2"},
+		},
+	}
+
+	// mock function response from listUsersWithContext(ctx)
+	minioListUsersMock = func() (map[string]madmin.UserInfo, error) {
+		return mockUserMap, nil
+	}
+	// get list users response this response should have Name, CreationDate, Size and Access
+	// as part of of each user
+	function := "listUsers()"
+	userMap, err := listUsers(adminClient)
+	if err != nil {
+		t.Errorf("Failed on %s:, error occurred: %s", function, err.Error())
+	}
+	// verify length of users is correct
+	assert.Equal(len(mockUserMap), len(userMap), fmt.Sprintf("Failed on %s: length of user's lists is not the same", function))
+
+	for _, b := range userMap {
+		assert.Contains(mockUserMap, b.AccessKey)
+		assert.Equal(string(mockUserMap[b.AccessKey].Status), b.Status)
```

Comment:
could you please also add the validation of `b.PolicyName` and probably changing the `status` of one to `disabled` to be different and ensure that both are tested against each other. And also if possible validate `MemberOf` probably using `
`assert. ElementsMatch()`

Thanks for adding tests 👍

Reply:
done, we don't have calls to update states on users yet, as soon as those are added I'll be putting a test for that